x-pack/docs/en/rest-api/security/get-tokens.asciidoc

```diff
@@ -63,9 +49,27 @@ pertain to creating a token:
 
 `grant_type`::
 (string) The type of grant. Supported grant types are: `password`, `_kerberos`,
-`client_credentials` and `refresh_token`. The `_kerberos` grant type
-is supported internally and implements SPNEGO based Kerberos support. The `_kerberos`
-grant type may change from version to version.
+`client_credentials` and `refresh_token`.
+
+`client_credentials`:::
+This grant type implements the Client Credentials Grant of OAuth2. It is geared
+for machine to machine communication and is not suitable or designed for the
+self-service user creation of tokens. It generates only access tokens that
+cannot be refreshed. The premise is that the entity that uses
+`client_credentials` has constant access to a set of (client, not end-user)
+credentials and can authenticate itself at will. 
+
+`_kerberos`::: 
+This grant type is supported internally and implements SPNEGO based Kerberos
+support. The `_kerberos` grant type may change from version to version.
+
+`password`:::
+This grant type implements the Resource Owner Password Credentials Grant of
+OAuth2. In this grant, a trusted client exchanges the end user's credentials
+for an access token and (possibly) a refresh token. The request needs to be made
+by an authenticated user but happens _on behalf_ of another authenticated user
+(the one whose credentials are passed as request parameters). This grant type is
+not suitable or designed for the self-service user creation of tokens.
 
 `password`::
```

Comment:
I'm not sure what is the best way to convey that these are `Required` but only when the appropriate grant is selected with the `(Optional/Required, <parameter_type> template. I can make arguments both for `Optional`, as in it is optional depending on the grant type parameter selected and for `Required` since these are required parameters for the relevant grant type. I'll defer to @lcawl when she is back from holidays to make the decision

Reply:
I chatted with the other folks working on this new API template and we propose using "Optional*" and clarify that it's required in XXX situation.  I've made that change in this PR.